aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1662,6 +1662,17 @@ static double std_var_all_cpu(const Tensor& self, double correction, bool take_s
   return result;
 }
 
+namespace {
+  static inline void warn_invalid_degrees_of_freedom(const char* fname, const TensorIterator iter, double correction) {
+    int64_t reducing_over_num_elements = iter.num_output_elements() == 0 ? 0 : iter.numel() / iter.num_output_elements();
+    if (reducing_over_num_elements - correction <= 0) {
+      std::ostringstream ss;
+      ss << fname << "(): degrees of freedom is <= 0";
```

Comment:
Should we include in the error message a way how the user can correct the problem?

Reply:
Right now, the warning text is the same as in `torch.cov`, but we can of course add something like: "correction should be less than the number of elements used in the reduction operation."

Reply:
That would be great. Maybe something like "correction should be strictly less than the reduction factor (input numel divided by output numel)".

Reply:
That would work for mean/std/var, but it's a bit more tricky for cov, because the reduction operations are implicitly used during the cov calculation.

```python3
>>> x = torch.randn(5, 10)
>>> x.numel()
50
>>> torch.cov(x).numel()
25
```

Correction should be fine up to `N-1 = 9` here.

Reply:
I used your proposed message for the normal reduction ops and provided a more specific message for `torch.cov` w.r.t. the number of observations.